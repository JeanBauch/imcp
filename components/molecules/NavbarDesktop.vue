<template>
	<nav class="hidden md:flex mx-auto">
		<ul class="flex justify-center items-center gap-7 xl:gap-12 2xl:gap-16 border-t-2 border-b-2 border-gray border-opacity-20 w-full">
			<li v-for="(item, key) in dataHeader" :key="key" class="text-white hover:text-primary-yellow-500">
				<router-link :to="{ name: item.link }">
					<h5 class="py-4 border-t-2 border-transparent hover:border-t-primary-yellow-500 expect-border-header">
						{{ item.name }}
					</h5>
				</router-link>
			</li>
		</ul>
	</nav>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
	data () {
		return {
			dataHeader: [
				{
					name: 'Início',
					link: 'index',
				},
				{
					name: 'Instituto',
					link: 'instituto',
				},
				{
					name: 'Escola de Música',
					link: 'utilites',
				},
				{
					name: 'Midia',
					link: 'utilites',
				},
				{
					name: 'Noticias',
					link: 'noticias',
				},
				{
					name: 'Agenda',
					link: 'utilites',
				},
				{
					name: 'Contato',
					link: 'contato',
				},
				{
					name: 'Transparência',
					link: 'utilites',
				},
			],
		};
	},
});
</script>
