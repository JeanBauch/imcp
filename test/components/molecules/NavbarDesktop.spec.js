import { shallowMount, RouterLinkStub } from '@vue/test-utils';

import NavbarDesktop from '~/components/molecules/NavbarDesktop.vue';

describe('Components:molecules:NavbarDesktop', () => {
	const factory = (values = {}) => {
		return shallowMount(NavbarDesktop, {
			data () {
				return {
					...values,
				};
			},
			stubs: {
				RouterLink: RouterLinkStub,
			},
		});
	};
	const arrayToDataHeader = [
		{
			name: 'Início',
			link: 'index',
		},
		{
			name: 'Instituto',
			link: 'utilites',
		},
		{
			name: 'Escola de Música',
			link: 'utilites',
		},
		{
			name: 'Midia',
			link: 'utilites',
		},
		{
			name: 'Noticias',
			link: 'noticias',
		},
		{
			name: 'Agenda',
			link: 'utilites',
		},
		{
			name: 'Contato',
			link: 'contato',
		},
		{
			name: 'Transparência',
			link: 'utilites',
		},
	];
	describe('Snapshots', () => {
		it('should render component', async () => {
			const wrapper = factory();

			await wrapper.setData({ dataHeader: arrayToDataHeader });

			expect(wrapper.html()).toMatchSnapshot();
		});
	});
});
