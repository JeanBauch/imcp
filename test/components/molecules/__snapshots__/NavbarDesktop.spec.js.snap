// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components:molecules:NavbarDesktop Snapshots should render component 1`] = `
"<nav class=\\"hidden md:flex mx-auto\\">
  <ul class=\\"flex justify-center items-center gap-7 xl:gap-12 2xl:gap-16 border-t-2 border-b-2 border-gray border-opacity-20 w-full\\">
    <li class=\\"text-white hover:text-primary-yellow-500\\"><a>
        <h5 class=\\"py-4 border-t-2 border-transparent hover:border-t-primary-yellow-500 expect-border-header\\">
          Início
        </h5>
      </a></li>
    <li class=\\"text-white hover:text-primary-yellow-500\\"><a>
        <h5 class=\\"py-4 border-t-2 border-transparent hover:border-t-primary-yellow-500 expect-border-header\\">
          Instituto
        </h5>
      </a></li>
    <li class=\\"text-white hover:text-primary-yellow-500\\"><a>
        <h5 class=\\"py-4 border-t-2 border-transparent hover:border-t-primary-yellow-500 expect-border-header\\">
          Escola de Música
        </h5>
      </a></li>
    <li class=\\"text-white hover:text-primary-yellow-500\\"><a>
        <h5 class=\\"py-4 border-t-2 border-transparent hover:border-t-primary-yellow-500 expect-border-header\\">
          Midia
        </h5>
      </a></li>
    <li class=\\"text-white hover:text-primary-yellow-500\\"><a>
        <h5 class=\\"py-4 border-t-2 border-transparent hover:border-t-primary-yellow-500 expect-border-header\\">
          Noticias
        </h5>
      </a></li>
    <li class=\\"text-white hover:text-primary-yellow-500\\"><a>
        <h5 class=\\"py-4 border-t-2 border-transparent hover:border-t-primary-yellow-500 expect-border-header\\">
          Agenda
        </h5>
      </a></li>
    <li class=\\"text-white hover:text-primary-yellow-500\\"><a>
        <h5 class=\\"py-4 border-t-2 border-transparent hover:border-t-primary-yellow-500 expect-border-header\\">
          Contato
        </h5>
      </a></li>
    <li class=\\"text-white hover:text-primary-yellow-500\\"><a>
        <h5 class=\\"py-4 border-t-2 border-transparent hover:border-t-primary-yellow-500 expect-border-header\\">
          Transparência
        </h5>
      </a></li>
  </ul>
</nav>"
`;
